hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFilePreadReader.java

```diff
@@ -40,8 +41,13 @@ public HFilePreadReader(ReaderContext context, HFileInfo fileInfo, CacheConfig c
 
     final MutableBoolean shouldCache = new MutableBoolean(true);
 
-    // Initialize HFileInfo object with metadata for caching decisions
-    fileInfo.initMetaAndIndex(this);
+    DataTieringManager dataTieringManager = DataTieringManager.getInstance();
+    if (dataTieringManager != null) {
+      // Initialize HFileInfo object with metadata for caching decisions.
+      // Initialize the metadata only if the data-tiering is enabled.
+      // If not, the metadata will be initialized later.
+      fileInfo.initMetaAndIndex(this);
+    }
```

Comment:
This check is to avoid the old functionality as is. We had observed that this line does introduce some performance regression leading to a unit test failure. We keep the old behaviour to initialise the metadata at the original location when data-tiering is not enabled.

Reply:
ok, I will remove the check here. The intention was to keep the old behaviour when the feature is disabled.